src/reducers/MapReducer.js

```diff
@@ -47,10 +54,79 @@ const INITIAL_STATE = {
     },
     mapLoaded: false,
     viewStyle: map.SearchOptions.MAP,
-    sites: []
+    sites: [],
+    displaySites: [],
+    filterCriteriaKeys: {accessibility: [], facilities: [], price: []}
 };
 
+const updateFilterKeys = ({filterCriteriaKeys}, filterKey) => {
+    let filterCriteriaSubKey = null;
+
+    if (_.includes(_.keys(accessibility_options), filterKey)) {
+        filterCriteriaSubKey = 'accessibility';
+    } else if (_.includes(_.keys(facilities_options), filterKey)) {
+        filterCriteriaSubKey = 'facilities';
+    } else if (_.includes(_.keys(price_options), filterKey)) {
+        filterCriteriaSubKey = 'price';
+    }
+
+    const keyIsAlreadyInList = _.includes(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    let updatedFilterKeyList = _.cloneDeep(filterCriteriaKeys);
+
+    if (keyIsAlreadyInList) {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.reject(filterCriteriaKeys[filterCriteriaSubKey], existingFilterKey => existingFilterKey === filterKey);
+    } else {
+        updatedFilterKeyList[filterCriteriaSubKey] = _.concat(filterCriteriaKeys[filterCriteriaSubKey], filterKey);
+    }
+
+    return updatedFilterKeyList;
+};
+
+const filterSites = ({sites}, updatedFilterKeys) => {
+    if (updatedFilterKeys.length === 0) {
+        return sites;
+    }
```

Comment:
If we don't have any filter keys, return early and just send back all the sites.